compiler/resolution/visibleFunctions.cpp

```diff
@@ -147,6 +181,155 @@ static void buildVisibleFunctionMap() {
+    bool fnBlock = false;
+    ModuleSymbol* inMod = block->getModule();
+    FnSymbol* inFn = block->getFunction();
+    BlockStmt* instantiationPt = NULL;
+    if (block->parentExpr != NULL) {
+      // not a module or function level block
+    } else if (inMod && block == inMod->block) {
+      moduleBlock = true;
+    } else if (inFn != NULL) {
+      INT_ASSERT(block->parentSymbol == inFn ||
+                 isArgSymbol(block->parentSymbol) ||
+                 isShadowVarSymbol(block->parentSymbol));
+      fnBlock = true;
+      BlockStmt* inFnInstantiationPoint = inFn->instantiationPoint();
+      if (inFnInstantiationPoint && !inFnInstantiationPoint->parentSymbol) {
+        INT_FATAL(inFn, "instantiation point not in tree\n"
+                        "try --break-on-remove-id %i and consider making\n"
+                        "that block scopeless",
+                        inFnInstantiationPoint->id);
+      }
+      if (inFnInstantiationPoint && inFnInstantiationPoint->parentSymbol)
+        instantiationPt = inFnInstantiationPoint;
+    }
+
+    if (call->id == breakOnResolveID) {
+      if (moduleBlock)
+        printf("visible methods: block %i  module %s  %s\n",
+               block->id, inMod->name, debugLoc(block));
+      else if (fnBlock)
+        printf("visible methods: block %i  fn %s  %s\n",
+               block->id, inFn->name, debugLoc(block));
+      else
+        printf("visible methods: block %i  %s\n",
+               block->id, debugLoc(block));
+
+      if (instantiationPt) {
+        printf("  instantiated from block %i  %s\n",
+               instantiationPt->id, debugLoc(instantiationPt));
+      }
+    }
+
+    // The following statement causes this to apply to all blocks,
+    // and not just module or function blocks.
+    //
+    // This is because e.g. in associative.chpl primer, instantiation occurs in
+    // a block that isn't a fn or module block.
+    visited.insert(block);
+
+    if (VisibleFunctionBlock* vfb = visibleFunctionMap.get(block)) {
+      // the block defines functions
+
+      if (Vec<FnSymbol*>* fns = vfb->visibleFunctions.get(name)) {
+        forv_Vec(FnSymbol, fn, *fns) {
+          // When private methods and fields are supported, we'll need to extend
+          // this
+          visibleFns.add(fn);
+        }
+      }
+    }
+
+    if (block->useList != NULL) {
+      // the block uses other modules
+      for_actuals(expr, block->useList) {
+        SymExpr* se = NULL;
+        if (UseStmt* use = toUseStmt(expr)) {
+          se = toSymExpr(use->src);
+        } else if (ImportStmt* import = toImportStmt(expr)) {
+          se = toSymExpr(import->src);
+        } else {
+          INT_FATAL("bad expr in useList, expected ImportStmt or UseStmt");
+        }
+        // Intentionally ignore use/import privacy and limitations
+        // Methods, fields, and special type support functions cannot be
+        // re-enabled via only lists (or doing so is not easy for the user).
```

Comment:
Do you mean renamed here?

Reply:
Nah, I mean that if the method is hidden because of an except list in one `use`, you can't explicitly include it in a different only list

Reply:
But methods can't be hidden in an except list at all, right?

Reply:
Right, sorry, meant to say only list in both sections.  Basically, if the symbol has been blanket excluded, we can't explicitly include it again